File: server/src/main/java/de/ketrwu/portfolio/forms/CaptchaForm.java
package de.ketrwu.portfolio.forms;

import de.ketrwu.portfolio.forms.validation.captcha.Captcha;
import lombok.Data;

import javax.validation.constraints.NotEmpty;

@Data
@Captcha(message = "Die Captcha-Lösung war nicht korrekt.")
public class CaptchaForm extends Form {

    @NotEmpty
    private String captchaResponse;

    @NotEmpty
    private String captchaToken;

    private String captchaImage;

}

File: server/src/main/kotlin/de/ketrwu/portfolio/forms/CaptchaForm.kt
package de.ketrwu.portfolio.forms

import de.ketrwu.portfolio.forms.validation.Captcha
import javax.validation.constraints.NotEmpty

/**
 * @author Kenneth Wußmann
 */

@Captcha(message = "Die Captcha-Lösung war nicht korrekt.")
open class CaptchaForm : Form() {

    @NotEmpty
    var captchaResponse: String? = null

    @NotEmpty
    var captchaToken: String? = null

    var captchaImage: String? = null
}
